core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_info.go

```diff
@@ -44,3 +46,14 @@ func getUpkeepType(id ocr2keepers.UpkeepIdentifier) upkeepType {
 	typeByte := id[upkeepTypeByteIndex]
 	return upkeepType(typeByte)
 }
+
+func UpkeepTriggerID(id *big.Int, trigger []byte) (string, error) {
+	idBytes := id.Bytes()
+
+	combined := append(idBytes, trigger...)
+
+	triggerIDBytes := crypto.Keccak256(combined)
+	triggerID := hex.EncodeToString(triggerIDBytes)
+
+	return triggerID, nil
+}
```

Comment:
a comment about where this is used / why it's needed might be helpful to future devs :)